lopezMidterm/src/components/Products.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import useFetch from '../hooks/useFetch';
+import './Products.css';
+
+interface Product {
+  id: number;
+  title: string;
+  image: string;
+}
+
+const Products: React.FC = () => {
+  const { data: products, loading, error } = useFetch<Product[]>('/products');
```

Comment:
if you hover over `products` it will be of type `Product[][]` which causes a lot of TS errors down below. To fix this just drop the array brackets after useFetch `useFetch<Product>`. That's because you told useFetch it's already returning an array.

This would be cleaner as it's own custom hook `useFetchProduct` and we'd return data as type products.